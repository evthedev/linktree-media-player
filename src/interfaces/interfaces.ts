import { DefaultColors, MusicProvider, MediaType } from '../common/constants';

export interface IRoute {
	path: string;
	exact: boolean;
	sidebar: string;
	icon: React.FC,
	main: any
}

export interface IMediaOwnProps {
	type: MediaType.Classic | MediaType.Music | MediaType.Show
}

export interface IMediaBaseProps {
	backgroundColor: string;
	fontColor: string;
}

export interface IClassic extends IMediaBaseProps {
	id: number;
	link: string
}

export interface IShow extends IMediaBaseProps {
	id: number;
	date: string;
	location: string;
	url: string;
	ticketsAvailable: number
}

export interface IMusic extends IMediaBaseProps {
	id: number;
	services: IService[]
}
export interface IService {
	id: number;
	providerName: MusicProvider;
	trackUrl: string
}

// export interface IAddress {
// 	id: number;
// 	quarantine_address: string;
// 	quarantine_address_lat: number | string;
// 	quarantine_address_long: number | string;
// }
// export interface ISanitisedAddress {
// 	id: number;
// 	quarantineAddress: string;
// 	quarantineAddressLat: number;
// 	quarantineAddressLong: number;
// }

// export interface ISanitisedUser {
// 	date: string;
// 	arrivedTravellers: number;
// 	totalUsersProvidedConsent: number;
// 	travellersArrivedAndProvidedConsent: number;
// 	travellersArrivedAndCompletedSetup: number;
// }

// export interface IPill {
// 	backgroundColor: DefaultColors
// }

export interface IState {
	links: IClassic[],
	shows: IShow[],
	music: IMusic[]
}

// export interface IMapProps {
// 	addresses: ISanitisedAddress[]
// }