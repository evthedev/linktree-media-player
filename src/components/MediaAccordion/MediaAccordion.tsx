import React from 'react';

import { Collapse } from 'antd';

import { IMusic, IShow, IMediaButtonOwnProps } from '../../interfaces/interfaces';
import { StyledPanel } from '../../common/styles';

export const MediaAccordion: React.FC<IMediaButtonOwnProps> = React.memo((props): JSX.Element => {

	const media = props.media as IMusic | IShow;
	
	return (
		<Collapse accordion={false} bordered={false} ghost={true} key={media.id} >
			<StyledPanel header={props.media.title} key={media.id} showArrow={false} {...props}>
				{ props.subComponent && props.subComponent(media) }
			</StyledPanel>
		</Collapse>
	)
});