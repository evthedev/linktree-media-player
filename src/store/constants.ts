import { IClassicLink, IShow, IMusic, IMediaBaseProps } from '../interfaces/interfaces';

export const SET_MEDIA_BUTTON_STYLES = 'SET_MEDIA_BUTTON_STYLES';
export const SET_MEDIA_DATA = 'SET_MEDIA_DATA';

export interface SetMediaDataAction {
	type: typeof SET_MEDIA_DATA;
	payload: IMusic[] | IShow[] | IClassicLink[]
}

export interface setMediaButtonStylesAction {
	type: typeof SET_MEDIA_BUTTON_STYLES;
	payload: IMediaBaseProps
}

export type actionTypes = SetMediaDataAction | setMediaButtonStylesAction