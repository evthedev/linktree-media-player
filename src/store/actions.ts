import { Dispatch } from 'react';
import { IClassicLink, IMediaBaseProps, IMusic, IShow } from '../interfaces/interfaces';
import { actionTypes, SET_MEDIA_BUTTON_STYLES, SET_MEDIA_DATA } from "./constants";
import { MediaType } from '../common/constants';
import { commandHandlers } from '../config/commandHandlers';

export const setMediaButtonStyles = (mediaButtonStyles: IMediaBaseProps): actionTypes => ({
	type: SET_MEDIA_BUTTON_STYLES,
	payload: mediaButtonStyles
})

export const setMediaData = (mediaData: IClassicLink[] | IShow[] | IMusic[]): actionTypes => ({
	type: SET_MEDIA_DATA,
	payload: mediaData
})

export const clearData = () => (
	(dispatch: Dispatch<actionTypes>) => {
		dispatch(setMediaData([]));
	}
)

export const fetchFakeMedia = (type: MediaType) => (
	async (dispatch: Dispatch<actionTypes>) => {
		// Get actionHandler for the specific media type
		
		// Get data for the specific media type
		const data = commandHandlers.filter(commandHandler => commandHandler.mediaType === type)[0].data;
		
		// Use a delay to simulate a delay when fetching from 'server'
		setTimeout(() => {
			dispatch(setMediaButtonStyles({
				backgroundColor: data.backgroundColor,
				fontColor: data.fontColor
			}))
			dispatch(setMediaData(data.results))
		}, 200)

	}
)
