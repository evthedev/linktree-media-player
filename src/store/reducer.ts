import { defaultState } from '../common/defaults';
import { IState } from '../interfaces/interfaces';
import { actionTypes, SET_MEDIA_BUTTON_STYLES, SET_MEDIA_DATA } from './constants';

export const reducer = (state: IState = defaultState, action: actionTypes): IState => {
	switch (action.type) {
		case SET_MEDIA_DATA:
			return {
				...state,
				mediaData: action.payload
			}	
		case SET_MEDIA_BUTTON_STYLES:
			return {
				...state,
				backgroundColor: action.payload.backgroundColor,
				fontColor: action.payload.fontColor
			}	
		default:
			return state		
	}
}
