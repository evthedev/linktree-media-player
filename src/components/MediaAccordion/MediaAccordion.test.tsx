import React from 'react';
import Enzyme, { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { MediaAccordion } from './MediaAccordion';
import { IState, IMediaButtonOwnProps } from '../../interfaces/interfaces';
import { createStore } from 'redux';
import { reducer } from '../../store/reducer';
import { Provider } from 'react-redux';

Enzyme.configure({ adapter: new EnzymeAdapter()});

describe('<MediaAccordion /> component', () => {
	const mockInitialState: IState = {
		backgroundColor: '',
		fontColor: '',
		mediaData: [],
	}

	const testMediaData: IMediaButtonOwnProps = {
		backgroundColor: 'azure',
		fontColor: 'aquamarine',
		media: {
			"id": 1,
			"title": "Test song 1",
			"services": [
				{
					"id": 1,
					"providerName": "Spotify",
					"trackUrl": "https://open.spotify.com/embed/track/sfasgasouytt74"
				},
			]
		}
	}

	const getWrapper = (mockStore = createStore(reducer, mockInitialState)) => mount(
		<Provider store={mockStore}>
			<MediaAccordion {...testMediaData} />
		</Provider>
	)
	it('should mount correctly with menu items', () => {
		const wrapper = getWrapper();
		expect(wrapper.isEmptyRender()).toBe(false);
		expect(wrapper).toMatchSnapshot();
	})
})