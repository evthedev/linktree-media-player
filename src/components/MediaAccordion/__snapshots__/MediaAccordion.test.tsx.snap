// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaAccordion /> component should mount correctly with menu items 1`] = `ReactWrapper {}`;
